src/commands/redis_cmd.cc

```diff
@@ -2270,6 +2277,54 @@ class CommandSInter : public Commander {
   }
 };
 
+class CommandSInterCard : public Commander {
+ public:
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    // args_ will be like: sintercard 2 key1 key2 LIMIT 1, where 2 is the number of keys, limit is optional
+    uint64_t num_keys = 0;
+    auto parse_num_keys = ParseInt<uint64_t>(args_[1], 10);
+    if (!parse_num_keys) {
+      return Status(Status::RedisParseErr, errValueNotInteger);
+    }
+    num_keys = *parse_num_keys;
+    if (num_keys <= 0) {
+      return Status(Status::NotOK, "numkeys should be greater than 0");
+    }
+    if (args_.size() < num_keys + 2) {
+      return Status(Status::NotOK, "numkeys should be equal to the number of keys");
+    }
+    if (args_.size() != num_keys + 2 && args_.size() != num_keys + 4) {
+      return Status(Status::NotOK, "invalid number of arguments");
+    }
+    int64_t limit = 0;
+    if (args_.size() == num_keys + 4) {
+      if (Util::ToLower(args_[num_keys + 2]) != "limit") {
+        return Status(Status::NotOK, "invalid number of arguments");
+      }
+      auto parse_limit = ParseInt<int64_t>(args_[num_keys + 3], 10);
+      if (!parse_limit) {
+        return Status(Status::RedisParseErr, errValueNotInteger);
+      }
+      limit = *parse_limit;
+      if (limit < 0) {
+        return Status(Status::NotOK, "limit can't be negative");
+      }
+    }
+    std::vector<Slice> keys;
+    for (uint64_t i = 2; i < num_keys + 2; i++) {
+      keys.emplace_back(args_[i]);
+    }
+    Redis::Set set_db(svr->storage_, conn->GetNamespace());
+    int64_t ret = 0;
+    auto s = set_db.InterCard(keys, limit, &ret);
+    if (!s.ok()) {
+      return Status(Status::RedisExecErr, s.ToString());
+    }
+    *output = Redis::Integer(ret);
+    return Status::OK();
+  }
+};
```

Comment:
You can refer to #1032  to use a more concise parsing approach.

Reply:
@tanruixiang Thanks for advice.

Reply:
It is optional to use the command parser, but I think at least we need extract the parsing part in this method `Execute` to `Parse`.